include/odp/api/spec/packet.h

```diff
@@ -1194,26 +1219,7 @@ typedef struct odp_packet_parse_param_t {
 	 *  layer. Checksum checking status can be queried for each packet with
 	 *  odp_packet_l3_chksum_status() and odp_packet_l4_chksum_status().
 	 */
-	union {
-		/** Individual check bits. */
-		struct {
-			/** Check IPv4 header checksum */
-			uint32_t ipv4_chksum   : 1;
-
-			/** Check UDP checksum */
-			uint32_t udp_chksum    : 1;
-
-			/** Check TCP checksum */
-			uint32_t tcp_chksum    : 1;
-
-			/** Check SCTP checksum */
-			uint32_t sctp_chksum   : 1;
-
-		} check;
-
-		/** All check bits. This can be used to set/clear all flags. */
-		uint32_t all_check;
-	};
+	odp_proto_chksums_t check;
```

Comment:
Sorry that I'm changing my mind here. But just realized that the original "check" bit-field was meant to be extendable with potential other checks, which are not only checksums but e.g. "check.ipv6.extension_xyz". So, this definition (check == chksums only) is unnecessary limiting. Could you change it to e.g.

odp_proto_chksums_t sums;

I think this is still readable:
parse.sums.all_chksum = 0;
parse.sums.chksum.udp = 1;
parse.sums.chksum.tcp = 1;

And then we are more future proof again:
parse.ipv6.all_ext = 0;
parse.ipv6.ext.mobile_ip = 1;

Reply:
@psavol changing to `checksums` (rather than just `sums`) name.

Reply:
Then it would need to be "chksums" per the type and chksum used in all other APIs (instead "checksum").

Note that the IPSEC param name needs the same update.

Reply:
done